src/main/scala/pureconfig/WordDelimiter.scala

```diff
@@ -0,0 +1,37 @@
+package pureconfig
+
+trait WordDelimiter {
+  def toTokens(s: String): Seq[String]
+  def fromTokens(l: Seq[String]): String
+}
+
+object CamelCaseWordDelimiter extends WordDelimiter {
+  def toTokens(s: String): Seq[String] = {
+    s.replaceAll(
```

Comment:
`replaceAll` recompiles the string into an argument each time it's called:
http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8-b132/java/lang/String.java#2209

I suggest compiling that regex once and storing it in a `private val`.